src/components/ArticleList.jsx

```diff
@@ -1,37 +1,58 @@
 import ArticleCard from "./ArticleCard"
-import axios from "axios"
 import React from "react"
 import {useEffect, useState} from "react"
 import api from "./Api"
+import { useNavigate, useParams } from "react-router-dom"
 
 export default function ArticleList () {
-    const [articlesList, setArticlesList] = useState([])
-    const [loading, setLoading] = useState(true)
+  const { topic } = useParams()
+  const [articlesList, setArticlesList] = useState([])
+  const [loading, setLoading] = useState(true)
+  const navigate = useNavigate()
 
 
-    useEffect(() => {
-        api.get("/articles")
-          .then((response) => {
-            setArticlesList(response.data.articles);
-            setLoading(false);
+  useEffect(() => {
+    const fetchArticles = () => {
+      api.get("/articles")
+        .then((response) => {
+          let filteredArticles = response.data.articles
+          if (topic) {
+            filteredArticles = filteredArticles.filter(article => article.topic === topic)
```

Comment:
If you finished task 11 on the back end you don't need to do a filter here and you can pass the topic as params to get articles which would return all of the articles by the topic anyway. If you didn't, this is a fine solution for now!